### PR TITLE
8263974: Move SystemDictionary::verify_protection_domain

### PR DESCRIPTION
**Pull request by coleenp**
Please review this mostly trivial fix to move SystemDictionary::validate_protection_domain into Dictionary and hide the functions in Dictionary that it calls.  This change also removes some #include dictionary.hpp and a TRAPS parameter where not needed.

See CR for more details.  This function was in the middle of others that I want to keep together in systemDictionary.cpp.

Tested with tier1 on 4 Oracle supported platforms.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8263974](https://bugs.openjdk.java.net/browse/JDK-8263974): Move SystemDictionary::verify_protection_domain

### Reviewers
 * [Harold Seigel](https://openjdk.java.net/census#hseigel) (@hseigel - **Reviewer**) ⚠️ Review applies to 33283cbf297f37f7c4ef163ede9bc5535e2d12bd
 * [Lois Foltan](https://openjdk.java.net/census#lfoltan) (@lfoltan - **Reviewer**)
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)

### Download
To checkout this PR locally:
`$ git fetch https://git.openjdk.java.net/jdk pull/3120/head:pull/3120`
`$ git checkout pull/3120`

To update a local copy of the PR:
`$ git checkout pull/3120`
`$ git pull https://git.openjdk.java.net/jdk pull/3120/head`

### CODE REVIEW

File: src/hotspot/share/classfile/dictionary.cpp [hunk 1]
```diff
@@ -341,14 +345,82 @@ void Dictionary::add_protection_domain(int index, unsigned int hash,
...
   return entry->is_valid_protection_domain(protection_domain);
 }
 
+void Dictionary::validate_protection_domain(unsigned int name_hash,
+                                            InstanceKlass* klass,
+                                            Handle class_loader,
+                                            Handle protection_domain,
+                                            TRAPS) {
+
+  // Now we have to call back to java to check if the initating class has access
```
Comment (hseigel) [inline src/hotspot/share/classfile/dictionary.cpp:362]: Can you move this comment down about 10 lines?

**Review by hseigel — APPROVED** [review]
Other than that one nit, this looks good.
Thanks, Harold

File: src/hotspot/share/classfile/dictionary.cpp (hunk 1 above)
Comment (coleenp) [reply src/hotspot/share/classfile/dictionary.cpp]: Ok Done.

**Review by lfoltan — APPROVED** [review]
Looks good!
Lois

**Comment by coleenp** [comment]
Thanks Harold and Lois!

**Review by dholmes-ora — APPROVED** [review]
Looks good!

Thanks,
David

**Comment by coleenp** [comment]
Thanks David!
/integrate

**Pull request closed without merging**